src/Google/AccessToken/Verify.php

```diff
@@ -0,0 +1,178 @@
+ *
+ */
+class Google_AccessToken_Verify
+{
+  const FEDERATED_SIGNON_CERT_URL = 'https://www.googleapis.com/oauth2/v3/certs';
+
+  /**
+   * @var GuzzleHttp\ClientInterface The http client
+   */
+  private $http;
+
+  /**
+   * @var Google\Auth\CacheInterface cache class
+   */
+  private $cache;
+
+  /**
+   * Instantiates the class, but does not initiate the login flow, leaving it
+   * to the discretion of the caller.
+   */
+  public function __construct(ClientInterface $http = null, CacheInterface $cache = null)
+  {
+    if (is_null($http)) {
+      $http = new Client();
+    }
+
+    $this->http = $http;
+    $this->cache = $cache;
+  }
+
+  /**
+   * Verifies an id token and returns the authenticated apiLoginTicket.
+   * Throws an exception if the id token is not valid.
+   * The audience parameter can be used to control which id tokens are
+   * accepted.  By default, the id token must have been issued to this OAuth2 client.
+   *
+   * @param $audience
+   * @return array the token payload, if successful
+   * @throws Google_Auth_Exception
+   */
+  public function verifyIdToken($idToken, $audience = null)
+  {
+    if (empty($idToken)) {
+      throw new LogicException('id_token cannot be null');
+    }
+
+    // Check signature
+    $certs = $this->getFederatedSignonCerts();
+    foreach ($certs as $cert) {
+      $modulus = new BigInteger(JWT::urlsafeB64Decode($cert['n']), 256);
+      $exponent = new BigInteger(JWT::urlsafeB64Decode($cert['e']), 256);
+
+      $rsa = new RSA();
+      $rsa->loadKey(array('n' => $modulus, 'e' => $exponent));
+
+      try {
+        $payload = JWT::decode(
+            $idToken,
+            $rsa->getPublicKey(),
+            array('RS256')
+        );
+        if (property_exists($payload, 'aud')) {
+          if ($audience && $payload->aud != $audience) {
+            return false;
+          }
+        }
+
+        return $payload;
+      } catch (ExpiredException $e) {
+        return false;
+      } catch (DomainException $e) {
+        // continue
+      }
+    }
+
+    return false;
+  }
+
+  private function getCache()
+  {
+    if (!$this->cache) {
+      $this->cache = $this->createDefaultCache();
+    }
+
+    return $this->cache;
+  }
+
+  private function createDefaultCache()
+  {
+    return new Google_Cache_File(
+        sys_get_temp_dir().'/google-api-php-client'
+    );
+  }
+
+  /**
+   * Retrieve and cache a certificates file.
+   *
+   * @param $url string location
+   * @throws Google_Auth_Exception
+   * @return array certificates
+   */
+  private function retrieveCertsFromLocation($url)
+  {
+    // If we're retrieving a local file, just grab it.
+    if ('http' != substr($url, 0, 4)) {
+      $file = file_get_contents($url);
+      if ($file) {
+        return json_decode($file, true);
+      } else {
+        throw new Google_Auth_Exception(
+            "Failed to retrieve verification certificates: '" .
+            $url . "'."
```

Comment:
yeah I don't like it either. It should be:

```php
sprintf('Failed to retrieve verification certificates: "%s".', $url)
```

and also at the top it would be better as


```php
if (0 !== strpos($url, 'http')) {
   //...
}
```